bigchaindb/common/transaction.py

```diff
@@ -768,20 +768,19 @@ def _sign_threshold_signature_fulfillment(self, input_, index,
                 key_pairs (dict): The keys to sign the Transaction with.
         """
         input_ = deepcopy(input_)
-        for owner_before in input_.owners_before:
-            try:
-                # TODO: CC should throw a KeypairMismatchException, instead of
-                #       our manual mapping here
-
-                # TODO FOR CC: Naming wise this is not so smart,
-                #              `get_subcondition` in fact doesn't return a
-                #              condition but a fulfillment
-
-                # TODO FOR CC: `get_subcondition` is singular. One would not
-                #              expect to get a list back.
-                ccffill = input_.fulfillment
-                subffill = ccffill.get_subcondition_from_vk(owner_before)[0]
-            except IndexError:
+        for owner_before in set(input_.owners_before):
+            # TODO: CC should throw a KeypairMismatchException, instead of
+            #       our manual mapping here
+
+            # TODO FOR CC: Naming wise this is not so smart,
+            #              `get_subcondition` in fact doesn't return a
+            #              condition but a fulfillment
+
+            # TODO FOR CC: `get_subcondition` is singular. One would not
+            #              expect to get a list back.
+            ccffill = input_.fulfillment
+            subffills = ccffill.get_subcondition_from_vk(owner_before)
+            if not subffills:
```

Comment:
This might be worth creating a ticket on Cryptoconditions that `get_subcondition_from_vk` should throw :thinking:
Thoughts?